### ISSUE TITLE
RFC4122 compatible binary representation for GUID

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/runtime#53354]
## Background and Motivation

Previous discussions: https://github.com/dotnet/runtime/issues/23868 https://github.com/dotnet/runtime/issues/29523
Currently, the `Guid` type uses "MS quirk" binary representation. It's hard to use it to communicate with RFC4122 compatible (all-BigEndian) representations.

## Proposed API

```diff
namespace System
{
    public struct Guid
    {
        public Guid(byte[]);
        public Guid(ReadOnlySpan<byte>);
+        public static Guid FromBigEndian(ReadOnlySpan<byte>);

        public byte[] ToByteArray();
+        public byte[] ToBigEndianByteArray();

        public bool TryWriteBytes(Span<byte>);
+        public bool TryWriteBigEndianBytes(Span<byte>);
    }
}
```

## Usage Examples

```csharp
Guid guid = new Guid("00112233-4455-6677-8899-aabbccddeeff");
var bytes = guid.ToBigEndianByteArray();
// Gets an array encodes bytes in the same order of string representation
```

## Alternative Designs

Use a new type duplicates the api surface of `Guid`.

## Risks

People may get confused about which representation to use.

**Comment by AaronRobinsonMSFT** [comment]
@GrabYourPitchforks I think you previously had some thoughts in this area. Perspective on this proposal?

**Comment by GrabYourPitchforks** [comment]
IMO this direction solves the heartache I had with https://github.com/dotnet/runtime/issues/23868. It's not adding new GUID factories, so it shouldn't contribute to significant consumer confusion. It's just allowing users to control how the structure is converted to / from a binary blob.

**Comment by danmoseley** [comment]
@GrabYourPitchforks  based on your comment above, should we mark as ready for review?

**Comment by tannergooding** [comment]
I wonder if users are going to expect a `TryWriteBigEndian` API to match `RFC4122` when the format of `Guid` itself is the Microsoft representation.

Having the name `TryWriteRfc4122Bytes` might be a "better" name in some sense due to that. Having a `BigEndian` API without a corresponding `LittleEndian` API is another concern.

**Comment by huoyaoyuan** [comment]
Should the name be discussed during API review? I'm somehow OK with `Rfc4122` in the name, but generally I haven't seen spec ids mentioned much in .NET APIs.

**Comment by huoyaoyuan** [comment]
Closed in favor of #86798.
